### ISSUE TITLE
distutils will be deprecated in Python 3.12

### ISSUE THREAD

**Issue by ljwolf** [pysal/pysal#1277]
[Across the project](https://github.com/search?q=org%3Apysal%20distutils&type=code), many packages use `distutils` for setting up installations. However, `distutils` will be [deprecated](https://peps.python.org/pep-0632/#backwards-compatibility). We will need to deal with this  by moving to [`pyproject.toml`](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/). The best way to do this is probably revise the submodule template to use the `pyproject.toml` specification, and then propagate that downwards. 

This will affect the metapackage, as the metapackage will be uninstallable on 3.12 until all its constituent parts are off of distutils.

**Comment by martinfleis** [comment]
We already started the transition in some packages. I think that the latest one was tobler (https://github.com/pysal/tobler/pull/174) but it seems that we will need to do this across all packages before next meta.

**Comment by ljwolf** [comment]
Yep, just logging it here because it will be blocking for release.

**Comment by jGaboardi** [comment]
#### Check packages for conformance

* [ ] `pysal/pysal`
* [ ] `pysal/access`
* [ ] `pysal/esda`
* [ ] `pysal/giddy`
* [ ] `pysal/inequality`
* [ ] `pysal/libpysal`
* [ ] `pysal/mapclassify`
* [ ] `pysal/mgwr`
* [ ] `pysal/momepy`
* [ ] `pysal/legendgram`
* [ ] `pysal/pointpats`
* [ ] `pysal/segregation`
* [ ] `pysal/spaghetti`
* [ ] `pysal/spglm`
* [ ] `pysal/spint`
* [ ] `pysal/splot`
* [ ] `pysal/spopt`
* [ ] `pysal/spreg`
* [ ] `pysal/spvcm`
* [ ] `pysal/tobler`
